__tests__/servertesting.test.js

```diff
@@ -41,12 +44,45 @@ describe("/get/api ", () => {
     return request(app)
       .get("/api")
       .expect(200)
-      .then((data) => {
-        data = JSON.parse(data.text);
-        console.log(data);
-        expect(data).toHaveProperty("GET /api");
-        expect(data).toHaveProperty("GET /api/topics");
-        expect(data).toHaveProperty("GET /api/articles");
+      .then(({ body }) => {
+        expect(body).toEqual(data);
+      });
+  });
+});
+describe("/get/api/articles/:article_id ", () => {
+  it("200: Should return an article object selected by a parametric endpoint, with apropriate keys", () => {
+    return request(app)
+      .get("/api/articles/2")
+      .expect(200)
+      .then(({ body }) => {
+        body = body.articles;
+        expect(body.length > 0);
+        body.forEach((body) => {
+          expect(body).toHaveProperty("author");
+          expect(body).toHaveProperty("title");
+          expect(body).toHaveProperty("article_id");
+          expect(body).toHaveProperty("body");
+          expect(body).toHaveProperty("topic");
+          expect(body).toHaveProperty("created_at");
+          expect(body).toHaveProperty("votes");
+          expect(body).toHaveProperty("article_img_url");
+        });
+      });
+  });
+  it("404: Should receive an approperiate error message for an incorrect value of the correct data type", () => {
+    return request(app)
+      .get("/api/articles/1000000")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toEqual("Page not found - invalid Id");
```

Comment:
I'd tweak the terminology here, in this example Id is valid, but it's not found
We usually call something that is the wrong data type invalid, whereas something that's valid but doesn't exist is not found